Source/SwiftLintFramework/Extensions/Configuration+Parsing.swift

```diff
@@ -0,0 +1,128 @@
+        func defaultStringArray(_ object: Any?) -> [String] {
+            return [String].array(of: object) ?? []
+        }
+
+        // Use either new 'opt_in_rules' or deprecated 'enabled_rules' for now.
+        let optInRules = defaultStringArray(
+            dict[Key.optInRules.rawValue] ?? dict[Key.enabledRules.rawValue]
+        )
+
+        // Log an error when supplying invalid keys in the configuration dictionary
+        let invalidKeys = Set(dict.keys).subtracting(Configuration.validKeys(ruleList: ruleList))
+        if !invalidKeys.isEmpty {
+            queuedPrintError("Configuration contains invalid keys:\n\(invalidKeys)")
+        }
+
+        let disabledRules = defaultStringArray(dict[Key.disabledRules.rawValue])
+        let whitelistRules = defaultStringArray(dict[Key.whitelistRules.rawValue])
+        let included = defaultStringArray(dict[Key.included.rawValue])
+        let excluded = defaultStringArray(dict[Key.excluded.rawValue])
+
+        Configuration.warnAboutDeprecations(configurationDictionary: dict, disabledRules: disabledRules,
+                                            optInRules: optInRules, whitelistRules: whitelistRules, ruleList: ruleList)
+
+        let configuredRules: [Rule]
+        do {
+            configuredRules = try ruleList.configuredRules(with: dict)
+        } catch RuleListError.duplicatedConfigurations(let ruleType) {
+            let aliases = ruleType.description.deprecatedAliases.map { "'\($0)'" }.joined(separator: ", ")
+            let identifier = ruleType.description.identifier
+            queuedPrintError("Multiple configurations found for '\(identifier)'. Check for any aliases: \(aliases).")
+            return nil
+        } catch {
+            return nil
+        }
+
+        self.init(disabledRules: disabledRules,
+                  optInRules: optInRules,
+                  enableAllRules: enableAllRules,
+                  whitelistRules: whitelistRules,
+                  included: included,
+                  excluded: excluded,
+                  warningThreshold: dict[Key.warningThreshold.rawValue] as? Int,
+                  reporter: dict[Key.reporter.rawValue] as? String ?? XcodeReporter.identifier,
+                  ruleList: ruleList,
+                  configuredRules: configuredRules,
+                  swiftlintVersion: dict[Key.swiftlintVersion.rawValue] as? String,
+                  cachePath: cachePath ?? dict[Key.cachePath.rawValue] as? String)
+    }
+
+    private static func warnAboutDeprecations(configurationDictionary dict: [String: Any],
+                                              disabledRules: [String] = [],
+                                              optInRules: [String] = [],
+                                              whitelistRules: [String] = [],
+                                              ruleList: RuleList) {
+
+        // Deprecation warning for "enabled_rules"
+        if dict[Key.enabledRules.rawValue] != nil {
```

Comment:
it's not a burden to maintain, so I'd rather keep it, especially since I think it's wise to keep support for the deprecation machinery up-to-date to keep it well maintained when we need to deprecate something again in the future.